crypto/mbedtls/Makefile

```diff
@@ -0,0 +1,133 @@
+MBEDTLS_ZIP = v$(MBEDTLS_VERSION).zip
+
+MBEDTLS_UNPACKNAME = mbedtls
+UNPACK ?= unzip -o
+
+MBEDTLS_UNPACKDIR =  $(WD)/$(MBEDTLS_UNPACKNAME)
+MBEDTLS_UNPACKLIBDIR = $(MBEDTLS_UNPACKDIR)/library
+
+# Help
+CFLAGS += -D__unix__
+CFLAGS += -I$(MBEDTLS_UNPACKDIR)/include
+
+# Crypto Source
+CSRCS += \
+    $(MBEDTLS_UNPACKLIBDIR)/aesni.c \
+    $(MBEDTLS_UNPACKLIBDIR)/arc4.c \
+    $(MBEDTLS_UNPACKLIBDIR)/aria.c \
+    $(MBEDTLS_UNPACKLIBDIR)/asn1parse.c \
+    $(MBEDTLS_UNPACKLIBDIR)/asn1write.c \
+    $(MBEDTLS_UNPACKLIBDIR)/base64.c \
+    $(MBEDTLS_UNPACKLIBDIR)/bignum.c \
+    $(MBEDTLS_UNPACKLIBDIR)/blowfish.c \
+    $(MBEDTLS_UNPACKLIBDIR)/camellia.c \
+    $(MBEDTLS_UNPACKLIBDIR)/ccm.c \
+    $(MBEDTLS_UNPACKLIBDIR)/chacha20.c \
+    $(MBEDTLS_UNPACKLIBDIR)/chachapoly.c \
+    $(MBEDTLS_UNPACKLIBDIR)/cipher.c \
+    $(MBEDTLS_UNPACKLIBDIR)/cipher_wrap.c \
+    $(MBEDTLS_UNPACKLIBDIR)/cmac.c \
+    $(MBEDTLS_UNPACKLIBDIR)/ctr_drbg.c \
+    $(MBEDTLS_UNPACKLIBDIR)/des.c \
+    $(MBEDTLS_UNPACKLIBDIR)/dhm.c \
+    $(MBEDTLS_UNPACKLIBDIR)/ecdh.c \
+    $(MBEDTLS_UNPACKLIBDIR)/ecdsa.c \
+    $(MBEDTLS_UNPACKLIBDIR)/ecjpake.c \
+    $(MBEDTLS_UNPACKLIBDIR)/ecp.c \
+    $(MBEDTLS_UNPACKLIBDIR)/ecp_curves.c \
+    $(MBEDTLS_UNPACKLIBDIR)/entropy.c \
+    $(MBEDTLS_UNPACKLIBDIR)/entropy_poll.c \
+    $(MBEDTLS_UNPACKLIBDIR)/error.c \
+    $(MBEDTLS_UNPACKLIBDIR)/gcm.c \
+    $(MBEDTLS_UNPACKLIBDIR)/havege.c \
+    $(MBEDTLS_UNPACKLIBDIR)/hkdf.c \
+    $(MBEDTLS_UNPACKLIBDIR)/hmac_drbg.c \
+    $(MBEDTLS_UNPACKLIBDIR)/md.c \
+    $(MBEDTLS_UNPACKLIBDIR)/md2.c \
+    $(MBEDTLS_UNPACKLIBDIR)/md4.c \
+    $(MBEDTLS_UNPACKLIBDIR)/md5.c \
+    $(MBEDTLS_UNPACKLIBDIR)/memory_buffer_alloc.c \
+    $(MBEDTLS_UNPACKLIBDIR)/nist_kw.c \
+    $(MBEDTLS_UNPACKLIBDIR)/oid.c \
+    $(MBEDTLS_UNPACKLIBDIR)/padlock.c \
+    $(MBEDTLS_UNPACKLIBDIR)/pem.c \
+    $(MBEDTLS_UNPACKLIBDIR)/pk.c \
+    $(MBEDTLS_UNPACKLIBDIR)/pk_wrap.c \
+    $(MBEDTLS_UNPACKLIBDIR)/pkcs12.c \
+    $(MBEDTLS_UNPACKLIBDIR)/pkcs5.c \
+    $(MBEDTLS_UNPACKLIBDIR)/pkparse.c \
+    $(MBEDTLS_UNPACKLIBDIR)/pkwrite.c \
+    $(MBEDTLS_UNPACKLIBDIR)/platform.c \
+    $(MBEDTLS_UNPACKLIBDIR)/platform_util.c \
+    $(MBEDTLS_UNPACKLIBDIR)/poly1305.c \
+    $(MBEDTLS_UNPACKLIBDIR)/psa_crypto.c \
+    $(MBEDTLS_UNPACKLIBDIR)/psa_crypto_driver_wrappers.c \
+    $(MBEDTLS_UNPACKLIBDIR)/psa_crypto_se.c \
+    $(MBEDTLS_UNPACKLIBDIR)/psa_crypto_slot_management.c \
+    $(MBEDTLS_UNPACKLIBDIR)/psa_crypto_storage.c \
+    $(MBEDTLS_UNPACKLIBDIR)/psa_its_file.c \
+    $(MBEDTLS_UNPACKLIBDIR)/ripemd160.c \
+    $(MBEDTLS_UNPACKLIBDIR)/rsa.c \
+    $(MBEDTLS_UNPACKLIBDIR)/rsa_internal.c \
+    $(MBEDTLS_UNPACKLIBDIR)/sha1.c \
+    $(MBEDTLS_UNPACKLIBDIR)/sha256.c \
+    $(MBEDTLS_UNPACKLIBDIR)/sha512.c \
+    $(MBEDTLS_UNPACKLIBDIR)/threading.c \
+    $(MBEDTLS_UNPACKLIBDIR)/timing.c \
+    $(MBEDTLS_UNPACKLIBDIR)/version.c \
+    $(MBEDTLS_UNPACKLIBDIR)/version_features.c \
+    $(MBEDTLS_UNPACKLIBDIR)/xtea.c \
```

Comment:
These are all the files?  are you planning the select some using Kconfig?
Otherwise we can grab all of them at once with `CSRCS := $(notdir $(wildcard $(MBEDTLS_SRCDIR)$(DELIM)*.c))`

Reply:
It's a subset of the files. Mbed TLS has three groups of files that are built that align with the libraries targets they create. I was not sure yet if that would be useful to keep.

Reply:
@Ouss4 I modified this a bit to be closer to what you recommended, does this look correct to you?

Reply:
yes, that should bring everything into the build.